examples/hello/CMakeLists.txt

```diff
@@ -3,31 +3,34 @@
 # accompanying file Copyright.txt for details.
 #------------------------------------------------------------------------------#
 
-add_subdirectory(bpAttributeWriter)
+add_subdirectory(bpAttributeWriteRead)
 add_subdirectory(bpFlushWriter)
 
 if(ADIOS2_HAVE_MPI)
   add_subdirectory(bpFWriteCRead)
 endif()
 
+if(ADIOS2_HAVE_SZ)
+  add_subdirectory(bpOperatorSZWriter)
+endif()
+
 add_subdirectory(bpReader)
-add_subdirectory(bpThreadWrite)
-add_subdirectory(bpTimeWriter)
-add_subdirectory(bpWriter)
 
+add_subdirectory(bpStepsWriteRead)
 if(ADIOS2_HAVE_CUDA OR ADIOS2_HAVE_Kokkos_CUDA)
-  add_subdirectory(bpWriteReadCuda)
+  add_subdirectory(bpStepsWriteReadCuda)
 endif()
-
 find_package(hip QUIET)
 if(ADIOS2_HAVE_Kokkos_HIP OR hip_FOUND)
-  add_subdirectory(bpWriteReadHip)
+  add_subdirectory(bpStepsWriteReadHip)
 endif()
```

Comment:
> Not to be addressed here, Note for another PR. 

For consistency across the CMake code we should not use find_package outside of detectoptions if within the project(adios2)